addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/handler/YeelightHandlerBase.java

```diff
@@ -259,23 +259,19 @@ public void onStatusChanged(String prop, DeviceStatus status) {
     }
 
     void updateBrightnessAndColorUI(DeviceStatus status) {
-        if (status.isPowerOff()) {
-            updateState(CHANNEL_BRIGHTNESS, new PercentType(0));
-        } else {
-            updateState(CHANNEL_BRIGHTNESS, new PercentType(status.getBrightness()));
-            HSBType hsbType = null;
-            if (status.getMode() == DeviceMode.MODE_COLOR) {
-                hsbType = HSBType.fromRGB(status.getR(), status.getG(), status.getB());
-            } else if (status.getMode() == DeviceMode.MODE_HSV) {
-                hsbType = new HSBType(new DecimalType(status.getHue()), new PercentType(status.getSat()),
-                        new PercentType(1));
-            }
-            if (hsbType != null) {
-                updateState(CHANNEL_COLOR, hsbType);
-            }
-            updateState(CHANNEL_COLOR_TEMPERATURE,
-                    new PercentType((status.getCt() - COLOR_TEMPERATURE_MINIMUM) / COLOR_TEMPERATURE_STEP));
-        }
+        PercentType brightness = status.isPowerOff() ? new PercentType(0) : new PercentType(status.getBrightness());
+
+        HSBType tempHsbType = HSBType.fromRGB(status.getR(), status.getG(), status.getB());
```

Comment:
Is this needed, or could this be done directly in the next check without the need of tempHsbType? Nonetheless this version might give it more structure, so just curious.

Reply:
It can be done without creating the tempHsbType, but it would need another IF check. Since the app only supports CT (Sunshine) and Color mode, the chances of HSV/HSB mode is very slim, so tempHsbType is most likely be used all the time.

Reply:
So you couldn't just replace the `new HSBType(tempHsbType.getHue(), tempHsbType.getSaturation(), brightness);` in the following lines with `new HSBType.fromRGB(status.getR(), status.getG(), status.getB());`?

Reply:
`HSBType.fromRGB(status.getR(), status.getG(), status.getB());` was what caused the problem if UI not updating correctly.

if you look at log from from the issue:
`status = DeviceStatus [isPowerOff=true, r=128, g=128, b=128, color=8421504, brightness=10, ct=5000, hue=0, sat=0, isFlowing=false, delayOff=-1, mFlowItems=[ColorFlowItem [duration=1996, mode=1, value=14438425, brightness=80], ColorFlowItem [duration=1996, mode=1, value=14448670, brightness=80], ColorFlowItem [duration=1996, mode=1, value=11153940, brightness=80]], mode=MODE_COLOR, isMusicOn=false, name=]`

Notice that `isPowerOff=true` and `r=128, g=128, b=128,` . If we just let RGB determine the brightness (which the `fromRGB `method does so), then it would incorrect becuase brigntness should be 0 when power is off, not 128. We have to determine the and set the brightness manually.

Reply:
Ah, now I see. Thanks for explaining!